### PR TITLE
Tab management

### PR DESCRIPTION
**Pull request by pkoryzna**
- closing tabs
- avoiding duplicates
- middle-click close :bomb:

### CODE REVIEW

File: frontend/src/main/scala/akkaviz/frontend/TabManager.scala [hunk 1]
```diff
@@ -1,15 +1,50 @@
...
+
+  def close(target: ClosableTab): Unit = {
+    if(target.isActive) activateSiblingOf(target)
+    target.tab.parentNode.removeChild(target.tab)
+    target.tabBody.parentNode.removeChild(target.tabBody)
+    target.onClose()
+    tabs.delete(target.tabId)
+  }
+
+  private def activateSiblingOf(ct: ClosableTab): Unit = {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/TabManager.scala]: Codacy Issue found: [activateSiblingOf should have a smaller scope by using private[this]](https://www.codacy.com/app/lustefaniak/akka-viz/file/2237743150/issues/source?bid=3121282&fileBranchId=3178508#l37)

File: frontend/src/main/scala/akkaviz/frontend/TabManager.scala [hunk 2]
```diff
@@ -1,15 +1,50 @@
...
+    Option(ct.tab.nextElementSibling).orElse(Option(ct.tab.previousElementSibling)).map { s =>
+      s.querySelector("a[data-toggle]").asInstanceOf[HTMLElement]
+    }.foreach{ _.click() }
+
+  }
+
+  private def handleMiddleClick(tab: ClosableTab)(e: MouseEvent): Unit = {
```
Comment (lustefaniak) [inline frontend/src/main/scala/akkaviz/frontend/TabManager.scala]: Codacy Issue found: [handleMiddleClick should have a smaller scope by using private[this]](https://www.codacy.com/app/lustefaniak/akka-viz/file/2237743150/issues/source?bid=3121282&fileBranchId=3178508#l44)

**Comment by JJag** [comment]
:ok_hand:

**Comment by lustefaniak** [comment]
:+1:

**Pull request merged**